path segment runtime error in root-level subdirs on linux/fish

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### What happened?

When in root-level subdirs (e.g. `/etc`, `/var`), the path segment throws a fatal runtime error. The root directory itself (`/`) works correctly, as do second-level and deeper directories (`/etc/fish`, `/var/log`). Note that I'm using the `powerline` style for paths, and based on the error it seems this may be related to the unique letter functionality.

I was able to reproduce this back to version 14.31.0 and #3731, with the last working version being 14.30.0.

### Theme

Custom:

```
#:schema https://raw.githubusercontent.com/JanDeDobbeleer/oh-my-posh/main/themes/schema.json

"$schema" = "https://raw.githubusercontent.com/JanDeDobbeleer/oh-my-posh/main/themes/schema.json"
final_space = true
version = 2.0

[[blocks]]
  alignment = "left"
  type = "prompt"
  newline = true

  [[blocks.segments]]
    background = "transparent"
    foreground = "blue"
    style = "plain"
    template = " {{ .Path }}"
    type = "path"
    [blocks.segments.properties]
      style = "powerlevel"

[[blocks]]
  alignment = "right"
  type = "prompt"

  [[blocks.segments]]
    background = "transparent"
    foreground = "#43CCEA"
    foreground_templates = ["{{ if or (.Working.Changed) (.Staging.Changed) }}#FF9248{{ end }}", "{{ if and (gt .Ahead 0) (gt .Behind 0) }}#ff4500{{ end }}", "{{ if gt .Ahead 0 }}#B388FF{{ end }}", "{{ if gt .Behind 0 }}#B388FF{{ end }}"]
    style = "plain"
    template = "{{ .UpstreamIcon }}{{ .HEAD }}{{if .BranchStatus }} {{ .BranchStatus }}{{ end }}{{ if .Working.Changed }} \uf044 {{ .Working.String }}{{ end }}{{ if and (.Working.Changed) (.Staging.Changed) }} |{{ end }}{{ if .Staging.Changed }} \uf046 {{ .Staging.String }}{{ end }}{{ if gt .StashCount 0 }} \uf692 {{ .StashCount }}{{ end }} "
    type = "git"
    [blocks.segments.properties]
      branch_max_length = 25.0
      fetch_stash_count = true
      fetch_status = true
      fetch_upstream_icon = true

  [[blocks.segments]]
    background = "transparent"
    foreground = "#E4F34A"
    style = "plain"
    template = "\ue235{{ if .Error }}{{ .Error }}{{ else }}{{ if .Venv }}{{ .Venv }} {{ end }}{{ .Full }}{{ end }} "
    type = "python"
    [blocks.segments.properties]
      fetch_version = false

  [[blocks.segments]]
    background = "transparent"
    foreground = "#7FD5EA"
    style = "plain"
    template = "\ue627{{ if .Error }}{{ .Error }}{{ else }}{{ .Full }}{{ end }} "
    type = "go"
    [blocks.segments.properties]
      fetch_version = true

  [[blocks.segments]]
    background = "transparent"
    foreground = "#42E66C"
    style = "plain"
    template = "\ue718{{ if .PackageManagerIcon }}{{ .PackageManagerIcon }} {{ end }}{{ .Full }} "
    type = "node"
    [blocks.segments.properties]
      fetch_version = true

  [[blocks.segments]]
    background = "transparent"
    foreground = "#E64747"
    style = "plain"
    template = "\ue791{{ if .Error }}{{ .Error }}{{ else }}{{ .Full }}{{ end }} "
    type = "ruby"
    [blocks.segments.properties]
      fetch_version = false

  [[blocks.segments]]
    background = "transparent"
    foreground = "#E64747"
    style = "plain"
    template = "\ue738{{ if .Error }}{{ .Error }}{{ else }}{{ .Full }}{{ end }} "
    type = "java"
    [blocks.segments.properties]
      fetch_version = false

  [[blocks.segments]]
    background = "transparent"
    foreground = "#9B6BDF"
    style = "plain"
    template = "\ue624{{ if .Error }}{{ .Error }}{{ else }}{{ .Full }}{{ end }} "
    type = "julia"
    [blocks.segments.properties]
      fetch_version = false

  [[blocks.segments]]
    background = "transparent"
    foreground = "#ffffff"
    style = "plain"
    template = "\ue73d{{ .Full }} "
    type = "php"
    [blocks.segments.properties]
      fetch_version = true

[[blocks]]
  alignment = "left"
  newline = true
  type = "prompt"

  [[blocks.segments]]
    background = "transparent"
    foreground = "#AEA4BF"
    style = "plain"
    template = "{{ .FormattedMs }} "
    type = "executiontime"
    [blocks.segments.properties]
      style = "austin"
      threshold = 150.0

  [[blocks.segments]]
    background = "transparent"
    foreground = "blue"
    style = "plain"
    template = "{{ if .SSHSession }}{{ .UserName }}@{{ .HostName }}{{end}}"
    type = "session"

  [[blocks.segments]]
    background = "transparent"
    foreground = "#E64747"
    foreground_templates = [
      "{{ if eq .Code 0 }}green{{ end }}"
    ]
    style = "plain"
    template = "❯"
    type = "text"

[transient_prompt]
  background = "transparent"
  foreground = "green"
  template = "❯ "
```

### What OS are you seeing the problem on?

Linux

### Which shell are you using?

fish

### Log output

```Shell
❯ oh-my-posh debug --plain

oh-my-posh fatal error rendering path segment:runtime error: index out of range [-1]

goroutine 1 [running]:
runtime/debug.Stack()
	/opt/hostedtoolcache/go/src/runtime/debug/stack.go:24 +0x65
github.com/jandedobbeleer/oh-my-posh/src/engine.(*Segment).SetEnabled.func1()
	/home/runner/work/oh-my-posh/oh-my-posh/src/engine/segment.go:467 +0x58
panic({0xc6a820, 0xc000634948})
	/opt/hostedtoolcache/go/src/runtime/panic.go:884 +0x213
github.com/jandedobbeleer/oh-my-posh/src/segments.(*Path).getUniqueLettersPath(0xc00017c980, 0x0)
	/home/runner/work/oh-my-posh/oh-my-posh/src/segments/path.go:352 +0x66f
github.com/jandedobbeleer/oh-my-posh/src/segments.(*Path).setStyle(0xc00017c980)
	/home/runner/work/oh-my-posh/oh-my-posh/src/segments/path.go:176 +0x365
github.com/jandedobbeleer/oh-my-posh/src/segments.(*Path).Enabled(0xc00017c980)
	/home/runner/work/oh-my-posh/oh-my-posh/src/segments/path.go:91 +0x3a
github.com/jandedobbeleer/oh-my-posh/src/engine.(*Segment).SetEnabled(0xc000602340, {0xe18a80, 0xc0002741c0})
	/home/runner/work/oh-my-posh/oh-my-posh/src/engine/segment.go:487 +0x177
github.com/jandedobbeleer/oh-my-posh/src/engine.(*Block).Debug(0xc000630000)
	/home/runner/work/oh-my-posh/oh-my-posh/src/engine/block.go:232 +0x1cf
github.com/jandedobbeleer/oh-my-posh/src/engine.(*Engine).PrintDebug(0xc00019fcd8, {0xcb31cc?, 0x0?, 0x187c1a0?}, {0xe0277c, 0x6})
	/home/runner/work/oh-my-posh/oh-my-posh/src/engine/debug.go:39 +0xb72
github.com/jandedobbeleer/oh-my-posh/src/cli.glob..func7(0x1854700?, {0xc000272520?, 0x1?, 0x1?})
	/home/runner/work/oh-my-posh/oh-my-posh/src/cli/debug.go:50 +0x327
github.com/spf13/cobra.(*Command).execute(0x1854700, {0xc000272500, 0x1, 0x1})
	/opt/hostedtoolcache/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:944 +0x847
github.com/spf13/cobra.(*Command).ExecuteC(0x1855560)
	/opt/hostedtoolcache/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:1068 +0x3bd
github.com/spf13/cobra.(*Command).Execute(...)
	/opt/hostedtoolcache/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:992
github.com/jandedobbeleer/oh-my-posh/src/cli.Execute({0xe0277c?, 0xc0000061a0?})
	/home/runner/work/oh-my-posh/oh-my-posh/src/cli/root.go:41 +0x4a
main.main()
	/home/runner/work/oh-my-posh/oh-my-posh/src/main.go:10 +0x27



Version: 15.3.0

Shell: fish (3.6.1)

Segments:

ConsoleTitle(false)                        -   0 ms - 
path(true)                                 -   0 ms -  
git(false)                                 -   0 ms - 
python(false)                              -   0 ms - 
go(false)                                  -   0 ms - 
node(false)                                -   0 ms - 
ruby(false)                                -   0 ms - 
java(false)                                -   0 ms - 
julia(false)                               -   0 ms - 
php(false)                                 -   0 ms - 
executiontime(false)                       -   0 ms - 
session(false)                             -   0 ms - 
text(true)                                 -   0 ms - ❯

Run duration: 7.154587ms

Cache path: /home/erwilson/.cache/oh-my-posh

Config path: /home/erwilson/.config/oh-my-posh.toml

Logs:

[DEBUG] 11:46:10.043 shell.go:Getenv:394 → NO DATA
[TRACE] 11:46:10.043 shell.go:Getenv(TERM_PROGRAM) - 16.978µs
[DEBUG] 11:46:10.043 shell.go:Getenv:394 → /home/erwilson/.cache
[TRACE] 11:46:10.043 shell.go:Getenv(XDG_CACHE_HOME) - 3.455µs
[TRACE] 11:46:10.043 shell_unix.go:CachePath() - 14.814µs
[DEBUG] 11:46:10.045 shell.go:Getenv:394 → /home/erwilson/.config/oh-my-posh.toml
[TRACE] 11:46:10.045 shell.go:Getenv(POSH_THEME) - 7.701µs
[DEBUG] 11:46:10.045 shell_unix.go:Platform:71 → fedora
[TRACE] 11:46:10.045 shell.go:resolveConfigPath() - 19.277µs
[TRACE] 11:46:10.045 shell.go:Init() - 2.331461ms
[TRACE] 11:46:10.045 shell.go:Flags() - 51ns
[TRACE] 11:46:10.047 config.go:loadConfig() - 1.700595ms
[TRACE] 11:46:10.047 shell.go:Flags() - 66ns
[DEBUG] 11:46:10.047 shell.go:Getenv:394 → NO DATA
[TRACE] 11:46:10.047 shell.go:Getenv(OMP_CACHE_DISABLED) - 3.415µs
[DEBUG] 11:46:10.048 shell.go:Shell:641 → process name: fish
[TRACE] 11:46:10.048 shell.go:Shell() - 1.069706ms
[DEBUG] 11:46:10.048 shell.go:Getenv:394 → 3.6.1
[TRACE] 11:46:10.048 shell.go:Getenv(POSH_SHELL_VERSION) - 3.185µs
[DEBUG] 11:46:10.048 debug.go:PrintDebug:22 → Segment: Title
[TRACE] 11:46:10.048 shell.go:Flags() - 76ns
[DEBUG] 11:46:10.048 block.go:Debug:230 → Segment: path
[TRACE] 11:46:10.048 shell.go:GOOS() - 65ns
[DEBUG] 11:46:10.048 shell.go:Pwd:424 → /var
[TRACE] 11:46:10.048 shell.go:Pwd() - 9.553µs
[TRACE] 11:46:10.048 shell.go:GOOS() - 28ns
[TRACE] 11:46:10.048 shell.go:Pwd() - 90ns
[TRACE] 11:46:10.048 shell.go:Shell() - 75ns
[TRACE] 11:46:10.048 shell.go:Shell() - 35ns
[DEBUG] 11:46:10.048 properties.go:GetBool:22 → mapped_locations_enabled: true
[DEBUG] 11:46:10.048 properties.go:GetString:28 → 
[DEBUG] 11:46:10.048 properties.go:GetString:28 → 
[TRACE] 11:46:10.048 shell.go:GOOS() - 28ns
[DEBUG] 11:46:10.048 properties.go:GetString:28 → ~
[DEBUG] 11:46:10.048 properties.go:GetKeyValueMap:46 → mapped_locations: map[]
[TRACE] 11:46:10.048 shell.go:PathSeparator() - 49ns
[TRACE] 11:46:10.048 shell.go:GOOS() - 37ns
[TRACE] 11:46:10.048 shell.go:GOOS() - 44ns
[TRACE] 11:46:10.048 shell.go:GOOS() - 35ns
[TRACE] 11:46:10.048 shell.go:GOOS() - 29ns
[TRACE] 11:46:10.048 shell.go:PathSeparator() - 33ns
[TRACE] 11:46:10.048 shell.go:PathSeparator() - 32ns
[TRACE] 11:46:10.048 shell.go:GOOS() - 28ns
[TRACE] 11:46:10.048 shell.go:GOOS() - 28ns
[TRACE] 11:46:10.048 shell.go:PathSeparator() - 27ns
[TRACE] 11:46:10.048 shell.go:GOOS() - 27ns
[TRACE] 11:46:10.048 shell.go:GOOS() - 28ns
[TRACE] 11:46:10.048 shell.go:PathSeparator() - 29ns
[TRACE] 11:46:10.048 shell.go:GOOS() - 35ns
[TRACE] 11:46:10.048 shell.go:GOOS() - 28ns
[TRACE] 11:46:10.048 shell.go:PathSeparator() - 29ns
[DEBUG] 11:46:10.048 properties.go:GetString:28 → powerlevel
[DEBUG] 11:46:10.048 properties.go:GetFloat64:34 → max_width: 0.000000
[DEBUG] 11:46:10.048 properties.go:GetString:28 → NO DATA
[TRACE] 11:46:10.048 shell.go:PathSeparator() - 29ns
[DEBUG] 11:46:10.048 properties.go:GetString:28 → /
[TRACE] 11:46:10.048 shell.go:PathSeparator() - 28ns
[TRACE] 11:46:10.048 shell.go:PathSeparator() - 28ns
[TRACE] 11:46:10.049 shell_unix.go:Root() - 465ns
[TRACE] 11:46:10.049 shell.go:Shell() - 50ns
[TRACE] 11:46:10.049 shell.go:ErrorCode() - 69ns
[DEBUG] 11:46:10.049 shell_unix.go:IsWsl:35 → false
[TRACE] 11:46:10.049 shell_unix.go:IsWsl() - 3.334µs
[TRACE] 11:46:10.049 shell.go:Pwd() - 141ns
[TRACE] 11:46:10.049 shell.go:GOOS() - 207ns
[TRACE] 11:46:10.049 shell.go:GOOS() - 28ns
[TRACE] 11:46:10.049 shell.go:GOOS() - 29ns
[TRACE] 11:46:10.049 shell.go:GOOS() - 28ns
[TRACE] 11:46:10.049 shell.go:GOOS() - 29ns
[DEBUG] 11:46:10.049 shell.go:User:545 → erwilson
[TRACE] 11:46:10.049 shell.go:User() - 2.94µs
[DEBUG] 11:46:10.049 shell.go:Host:557 → r9133vpk
[TRACE] 11:46:10.049 shell.go:Host() - 3.64µs
[TRACE] 11:46:10.049 shell.go:GOOS() - 28ns
[DEBUG] 11:46:10.049 shell_unix.go:Platform:71 → fedora
[DEBUG] 11:46:10.049 shell.go:Getenv:394 → 1
[TRACE] 11:46:10.049 shell.go:Getenv(SHLVL) - 2.444µs
[TRACE] 11:46:10.049 shell.go:TemplateCache() - 96.704µs
[TRACE] 11:46:10.049 shell.go:Shell() - 45ns
[TRACE] 11:46:10.049 shell.go:Flags() - 72ns
[DEBUG] 11:46:10.049 block.go:Debug:230 → Segment: git
[TRACE] 11:46:10.049 shell.go:Pwd() - 66ns
[TRACE] 11:46:10.049 shell.go:GOOS() - 27ns
[DEBUG] 11:46:10.049 shell_unix.go:IsWsl:35 → false
[TRACE] 11:46:10.049 shell_unix.go:IsWsl() - 2.664µs
[TRACE] 11:46:10.049 shell_unix.go:IsWsl2() - 5.142µs
[TRACE] 11:46:10.049 shell.go:GOOS() - 28ns
[DEBUG] 11:46:10.049 shell.go:CommandPath:596 → /usr/bin/git
[TRACE] 11:46:10.049 shell.go:CommandPath(git) - 34.823µs
[TRACE] 11:46:10.049 shell.go:HasCommand(git) - 40.431µs
[TRACE] 11:46:10.049 shell.go:Pwd() - 46ns
[ERROR] 11:46:10.049 shell.go:HasParentFilePath:729 → stat .git: no such file or directory
[TRACE] 11:46:10.049 shell.go:HasParentFilePath(.git) - 11.085µs
[DEBUG] 11:46:10.049 properties.go:GetBool:22 → fetch_bare_info: false
[DEBUG] 11:46:10.049 block.go:Debug:230 → Segment: python
[DEBUG] 11:46:10.049 properties.go:GetString:28 → environment
[TRACE] 11:46:10.049 shell.go:Pwd() - 44ns
[TRACE] 11:46:10.049 shell.go:GOOS() - 26ns
[DEBUG] 11:46:10.049 properties.go:GetStringArray:52 → extensions: [*.py *.ipynb pyproject.toml venv.bak]
[DEBUG] 11:46:10.049 properties.go:GetStringArray:52 → folders: [.venv venv virtualenv env venv-win pyenv-win]
[DEBUG] 11:46:10.049 properties.go:GetBool:22 → home_enabled: false
[TRACE] 11:46:10.049 shell.go:Pwd() - 43ns
[DEBUG] 11:46:10.049 properties.go:GetBool:22 → fetch_virtual_env: true
[DEBUG] 11:46:10.049 shell.go:Getenv:394 → NO DATA
[TRACE] 11:46:10.049 shell.go:Getenv(VIRTUAL_ENV) - 2.152µs
[DEBUG] 11:46:10.049 shell.go:Getenv:394 → NO DATA
[TRACE] 11:46:10.049 shell.go:Getenv(CONDA_ENV_PATH) - 1.96µs
[DEBUG] 11:46:10.049 shell.go:Getenv:394 → NO DATA
[TRACE] 11:46:10.049 shell.go:Getenv(CONDA_DEFAULT_ENV) - 1.88µs
[DEBUG] 11:46:10.049 block.go:Debug:230 → Segment: go
[TRACE] 11:46:10.049 shell.go:Pwd() - 43ns
[TRACE] 11:46:10.049 shell.go:GOOS() - 29ns
[DEBUG] 11:46:10.049 properties.go:GetStringArray:52 → extensions: [*.go go.mod]
[DEBUG] 11:46:10.049 properties.go:GetStringArray:52 → folders: []
[DEBUG] 11:46:10.049 properties.go:GetBool:22 → home_enabled: false
[TRACE] 11:46:10.049 shell.go:Pwd() - 45ns
[DEBUG] 11:46:10.049 properties.go:GetString:28 → files
[TRACE] 11:46:10.049 shell.go:Pwd() - 44ns
[DEBUG] 11:46:10.049 shell.go:HasFiles:446 → false
[TRACE] 11:46:10.049 shell.go:HasFiles(*.go) - 52.759µs
[TRACE] 11:46:10.049 shell.go:Pwd() - 70ns
[DEBUG] 11:46:10.049 shell.go:HasFiles:446 → false
[TRACE] 11:46:10.049 shell.go:HasFiles(go.mod) - 8.172µs
[DEBUG] 11:46:10.049 block.go:Debug:230 → Segment: node
[TRACE] 11:46:10.049 shell.go:Pwd() - 44ns
[TRACE] 11:46:10.049 shell.go:GOOS() - 28ns
[DEBUG] 11:46:10.049 properties.go:GetStringArray:52 → extensions: [*.js *.ts package.json .nvmrc pnpm-workspace.yaml .pnpmfile.cjs .npmrc .vue]
[DEBUG] 11:46:10.049 properties.go:GetStringArray:52 → folders: []
[DEBUG] 11:46:10.049 properties.go:GetBool:22 → home_enabled: false
[TRACE] 11:46:10.049 shell.go:Pwd() - 44ns
[DEBUG] 11:46:10.049 properties.go:GetString:28 → files
[DEBUG] 11:46:10.049 properties.go:GetBool:22 → fetch_package_manager: false
[TRACE] 11:46:10.049 shell.go:Pwd() - 44ns
[DEBUG] 11:46:10.049 shell.go:HasFiles:446 → false
[TRACE] 11:46:10.049 shell.go:HasFiles(*.js) - 30.72µs
[TRACE] 11:46:10.049 shell.go:Pwd() - 83ns
[DEBUG] 11:46:10.049 shell.go:HasFiles:446 → false
[TRACE] 11:46:10.049 shell.go:HasFiles(*.ts) - 25.324µs
[TRACE] 11:46:10.049 shell.go:Pwd() - 56ns
[DEBUG] 11:46:10.049 shell.go:HasFiles:446 → false
[TRACE] 11:46:10.049 shell.go:HasFiles(package.json) - 7.412µs
[TRACE] 11:46:10.049 shell.go:Pwd() - 44ns
[DEBUG] 11:46:10.049 shell.go:HasFiles:446 → false
[TRACE] 11:46:10.049 shell.go:HasFiles(.nvmrc) - 9.454µs
[TRACE] 11:46:10.049 shell.go:Pwd() - 43ns
[DEBUG] 11:46:10.049 shell.go:HasFiles:446 → false
[TRACE] 11:46:10.049 shell.go:HasFiles(pnpm-workspace.yaml) - 6.475µs
[TRACE] 11:46:10.049 shell.go:Pwd() - 44ns
[DEBUG] 11:46:10.049 shell.go:HasFiles:446 → false
[TRACE] 11:46:10.049 shell.go:HasFiles(.pnpmfile.cjs) - 6.265µs
[TRACE] 11:46:10.049 shell.go:Pwd() - 44ns
[DEBUG] 11:46:10.049 shell.go:HasFiles:446 → false
[TRACE] 11:46:10.049 shell.go:HasFiles(.npmrc) - 6.114µs
[TRACE] 11:46:10.049 shell.go:Pwd() - 43ns
[DEBUG] 11:46:10.049 shell.go:HasFiles:446 → false
[TRACE] 11:46:10.049 shell.go:HasFiles(.vue) - 6.976µs
[DEBUG] 11:46:10.049 block.go:Debug:230 → Segment: ruby
[TRACE] 11:46:10.049 shell.go:Pwd() - 43ns
[TRACE] 11:46:10.049 shell.go:GOOS() - 35ns
[DEBUG] 11:46:10.049 properties.go:GetStringArray:52 → extensions: [*.rb Rakefile Gemfile]
[DEBUG] 11:46:10.049 properties.go:GetStringArray:52 → folders: []
[DEBUG] 11:46:10.049 properties.go:GetBool:22 → home_enabled: false
[TRACE] 11:46:10.049 shell.go:Pwd() - 44ns
[DEBUG] 11:46:10.049 properties.go:GetString:28 → files
[TRACE] 11:46:10.049 shell.go:Pwd() - 44ns
[DEBUG] 11:46:10.049 shell.go:HasFiles:446 → false
[TRACE] 11:46:10.049 shell.go:HasFiles(*.rb) - 26.016µs
[TRACE] 11:46:10.049 shell.go:Pwd() - 64ns
[DEBUG] 11:46:10.049 shell.go:HasFiles:446 → false
[TRACE] 11:46:10.049 shell.go:HasFiles(Rakefile) - 7.036µs
[TRACE] 11:46:10.049 shell.go:Pwd() - 42ns
[DEBUG] 11:46:10.049 shell.go:HasFiles:446 → false
[TRACE] 11:46:10.049 shell.go:HasFiles(Gemfile) - 11.205µs
[DEBUG] 11:46:10.049 block.go:Debug:230 → Segment: java
[DEBUG] 11:46:10.049 shell.go:Getenv:394 → NO DATA
[TRACE] 11:46:10.049 shell.go:Getenv(JAVA_HOME) - 2.337µs
[TRACE] 11:46:10.049 shell.go:Pwd() - 52ns
[TRACE] 11:46:10.049 shell.go:GOOS() - 30ns
[DEBUG] 11:46:10.049 properties.go:GetStringArray:52 → extensions: [pom.xml build.gradle.kts build.sbt .java-version .deps.edn project.clj build.boot *.java *.class *.gradle *.jar *.clj *.cljc]
[DEBUG] 11:46:10.049 properties.go:GetStringArray:52 → folders: []
[DEBUG] 11:46:10.049 properties.go:GetBool:22 → home_enabled: false
[TRACE] 11:46:10.049 shell.go:Pwd() - 43ns
[DEBUG] 11:46:10.049 properties.go:GetString:28 → files
[TRACE] 11:46:10.049 shell.go:Pwd() - 42ns
[DEBUG] 11:46:10.049 shell.go:HasFiles:446 → false
[TRACE] 11:46:10.049 shell.go:HasFiles(pom.xml) - 6.668µs
[TRACE] 11:46:10.049 shell.go:Pwd() - 44ns
[DEBUG] 11:46:10.049 shell.go:HasFiles:446 → false
[TRACE] 11:46:10.049 shell.go:HasFiles(build.gradle.kts) - 6.278µs
[TRACE] 11:46:10.049 shell.go:Pwd() - 43ns
[DEBUG] 11:46:10.049 shell.go:HasFiles:446 → false
[TRACE] 11:46:10.049 shell.go:HasFiles(build.sbt) - 8.65µs
[TRACE] 11:46:10.049 shell.go:Pwd() - 43ns
[DEBUG] 11:46:10.049 shell.go:HasFiles:446 → false
[TRACE] 11:46:10.049 shell.go:HasFiles(.java-version) - 6.752µs
[TRACE] 11:46:10.049 shell.go:Pwd() - 45ns
[DEBUG] 11:46:10.049 shell.go:HasFiles:446 → false
[TRACE] 11:46:10.049 shell.go:HasFiles(.deps.edn) - 6.187µs
[TRACE] 11:46:10.049 shell.go:Pwd() - 43ns
[DEBUG] 11:46:10.049 shell.go:HasFiles:446 → false
[TRACE] 11:46:10.049 shell.go:HasFiles(project.clj) - 6.127µs
[TRACE] 11:46:10.049 shell.go:Pwd() - 43ns
[DEBUG] 11:46:10.049 shell.go:HasFiles:446 → false
[TRACE] 11:46:10.049 shell.go:HasFiles(build.boot) - 7.531µs
[TRACE] 11:46:10.049 shell.go:Pwd() - 44ns
[DEBUG] 11:46:10.049 shell.go:HasFiles:446 → false
[TRACE] 11:46:10.049 shell.go:HasFiles(*.java) - 32.382µs
[TRACE] 11:46:10.049 shell.go:Pwd() - 85ns
[DEBUG] 11:46:10.049 shell.go:HasFiles:446 → false
[TRACE] 11:46:10.049 shell.go:HasFiles(*.class) - 31.214µs
[TRACE] 11:46:10.049 shell.go:Pwd() - 72ns
[DEBUG] 11:46:10.049 shell.go:HasFiles:446 → false
[TRACE] 11:46:10.049 shell.go:HasFiles(*.gradle) - 27.383µs
[TRACE] 11:46:10.049 shell.go:Pwd() - 84ns
[DEBUG] 11:46:10.049 shell.go:HasFiles:446 → false
[TRACE] 11:46:10.049 shell.go:HasFiles(*.jar) - 23.365µs
[TRACE] 11:46:10.049 shell.go:Pwd() - 55ns
[DEBUG] 11:46:10.049 shell.go:HasFiles:446 → false
[TRACE] 11:46:10.049 shell.go:HasFiles(*.clj) - 23.961µs
[TRACE] 11:46:10.049 shell.go:Pwd() - 73ns
[DEBUG] 11:46:10.049 shell.go:HasFiles:446 → false
[TRACE] 11:46:10.049 shell.go:HasFiles(*.cljc) - 24.519µs
[DEBUG] 11:46:10.049 block.go:Debug:230 → Segment: julia
[TRACE] 11:46:10.049 shell.go:Pwd() - 102ns
[TRACE] 11:46:10.049 shell.go:GOOS() - 36ns
[DEBUG] 11:46:10.049 properties.go:GetStringArray:52 → extensions: [*.jl]
[DEBUG] 11:46:10.049 properties.go:GetStringArray:52 → folders: []
[DEBUG] 11:46:10.049 properties.go:GetBool:22 → home_enabled: false
[TRACE] 11:46:10.049 shell.go:Pwd() - 43ns
[DEBUG] 11:46:10.049 properties.go:GetString:28 → files
[TRACE] 11:46:10.049 shell.go:Pwd() - 42ns
[DEBUG] 11:46:10.050 shell.go:HasFiles:446 → false
[TRACE] 11:46:10.050 shell.go:HasFiles(*.jl) - 29.438µs
[DEBUG] 11:46:10.050 block.go:Debug:230 → Segment: php
[TRACE] 11:46:10.050 shell.go:Pwd() - 85ns
[TRACE] 11:46:10.050 shell.go:GOOS() - 30ns
[DEBUG] 11:46:10.050 properties.go:GetStringArray:52 → extensions: [*.php composer.json composer.lock .php-version blade.php]
[DEBUG] 11:46:10.050 properties.go:GetStringArray:52 → folders: []
[DEBUG] 11:46:10.050 properties.go:GetBool:22 → home_enabled: false
[TRACE] 11:46:10.050 shell.go:Pwd() - 43ns
[DEBUG] 11:46:10.050 properties.go:GetString:28 → files
[TRACE] 11:46:10.050 shell.go:Pwd() - 44ns
[DEBUG] 11:46:10.050 shell.go:HasFiles:446 → false
[TRACE] 11:46:10.050 shell.go:HasFiles(*.php) - 25.589µs
[TRACE] 11:46:10.050 shell.go:Pwd() - 69ns
[DEBUG] 11:46:10.050 shell.go:HasFiles:446 → false
[TRACE] 11:46:10.050 shell.go:HasFiles(composer.json) - 10.583µs
[TRACE] 11:46:10.050 shell.go:Pwd() - 43ns
[DEBUG] 11:46:10.050 shell.go:HasFiles:446 → false
[TRACE] 11:46:10.050 shell.go:HasFiles(composer.lock) - 6.163µs
[TRACE] 11:46:10.050 shell.go:Pwd() - 42ns
[DEBUG] 11:46:10.050 shell.go:HasFiles:446 → false
[TRACE] 11:46:10.050 shell.go:HasFiles(.php-version) - 6.003µs
[TRACE] 11:46:10.050 shell.go:Pwd() - 45ns
[DEBUG] 11:46:10.050 shell.go:HasFiles:446 → false
[TRACE] 11:46:10.050 shell.go:HasFiles(blade.php) - 6.067µs
[TRACE] 11:46:10.050 shell.go:Flags() - 50ns
[DEBUG] 11:46:10.050 block.go:Debug:230 → Segment: executiontime
[TRACE] 11:46:10.050 shell.go:Pwd() - 45ns
[TRACE] 11:46:10.050 shell.go:GOOS() - 27ns
[DEBUG] 11:46:10.050 properties.go:GetBool:22 → always_enabled: false
[TRACE] 11:46:10.050 shell.go:ExecutionTime() - 69ns
[DEBUG] 11:46:10.050 properties.go:GetFloat64:34 → threshold: 150.000000
[DEBUG] 11:46:10.050 block.go:Debug:230 → Segment: session
[TRACE] 11:46:10.050 shell.go:Pwd() - 44ns
[TRACE] 11:46:10.050 shell.go:GOOS() - 28ns
[DEBUG] 11:46:10.050 shell.go:Getenv:394 → NO DATA
[TRACE] 11:46:10.050 shell.go:Getenv(SSH_CONNECTION) - 2.106µs
[DEBUG] 11:46:10.050 shell.go:Getenv:394 → NO DATA
[TRACE] 11:46:10.050 shell.go:Getenv(SSH_CLIENT) - 9.165µs
[TRACE] 11:46:10.050 shell.go:TemplateCache() - 78ns
[TRACE] 11:46:10.050 shell.go:TemplateCache() - 74ns
[TRACE] 11:46:10.050 shell.go:TemplateCache() - 28ns
[TRACE] 11:46:10.050 shell.go:Shell() - 58ns
[DEBUG] 11:46:10.050 block.go:Debug:230 → Segment: text
[TRACE] 11:46:10.050 shell.go:Pwd() - 72ns
[TRACE] 11:46:10.050 shell.go:GOOS() - 28ns
[TRACE] 11:46:10.050 shell.go:TemplateCache() - 34ns
[TRACE] 11:46:10.050 shell.go:Shell() - 34ns
[TRACE] 11:46:10.050 shell.go:TemplateCache() - 115ns
[DEBUG] 11:46:10.050 shell.go:Getenv:394 → /home/erwilson/.cache
[TRACE] 11:46:10.050 shell.go:Getenv(XDG_CACHE_HOME) - 4.331µs
[TRACE] 11:46:10.050 shell_unix.go:CachePath() - 15.381µs
[TRACE] 11:46:10.050 shell.go:Flags() - 62ns

oh-my-posh fatal error rendering path segment:runtime error: index out of range [-1]

goroutine 36 [running]:
runtime/debug.Stack()
	/opt/hostedtoolcache/go/src/runtime/debug/stack.go:24 +0x65
github.com/jandedobbeleer/oh-my-posh/src/engine.(*Segment).SetEnabled.func1()
	/home/runner/work/oh-my-posh/oh-my-posh/src/engine/segment.go:467 +0x58
panic({0xc6a820, 0xc00014a270})
	/opt/hostedtoolcache/go/src/runtime/panic.go:884 +0x213
github.com/jandedobbeleer/oh-my-posh/src/segments.(*Path).getUniqueLettersPath(0xc00017c600, 0x0)
	/home/runner/work/oh-my-posh/oh-my-posh/src/segments/path.go:352 +0x66f
github.com/jandedobbeleer/oh-my-posh/src/segments.(*Path).setStyle(0xc00017c600)
	/home/runner/work/oh-my-posh/oh-my-posh/src/segments/path.go:176 +0x365
github.com/jandedobbeleer/oh-my-posh/src/segments.(*Path).Enabled(0xc00017c600)
	/home/runner/work/oh-my-posh/oh-my-posh/src/segments/path.go:91 +0x3a
github.com/jandedobbeleer/oh-my-posh/src/engine.(*Segment).SetEnabled(0xc0000c4b60, {0xe18a80, 0xc000274230})
	/home/runner/work/oh-my-posh/oh-my-posh/src/engine/segment.go:487 +0x177
github.com/jandedobbeleer/oh-my-posh/src/engine.(*Block).setEnabledSegments.func1(0x0?)
	/home/runner/work/oh-my-posh/oh-my-posh/src/engine/block.go:112 +0x6a
created by github.com/jandedobbeleer/oh-my-posh/src/engine.(*Block).setEnabledSegments
	/home/runner/work/oh-my-posh/oh-my-posh/src/engine/block.go:110 +0x98
```

--- a/src/segments/path.go
+++ b/src/segments/path.go
@@ -249,7 +249,10 @@ func (pt *Path) getAgnosterPath() string {
 	for i := 1; i < n; i++ {
 		elements = append(elements, folderIcon)
 	}
-	elements = append(elements, splitted[n-1])
+
+	if len(splitted) > 0 {
+		elements = append(elements, splitted[n-1])
+	}
 
 	return pt.colorizePath(pt.root, elements)
 }
@@ -299,7 +302,10 @@ func (pt *Path) getLetterPath() string {
 		letter := pt.getRelevantLetter(splitted[i])
 		elements = append(elements, letter)
 	}
-	elements = append(elements, splitted[n-1])
+
+	if len(splitted) > 0 {
+		elements = append(elements, splitted[n-1])
+	}
 
 	return pt.colorizePath(pt.root, elements)
 }
@@ -349,7 +355,10 @@ func (pt *Path) getUniqueLettersPath(maxWidth int) string {
 			}
 		}
 	}
-	elements = append(elements, splitted[n-1])
+
+	if len(elements) > 0 {
+		elements = append(elements, splitted[n-1])
+	}
 
 	return pt.colorizePath(pt.root, elements)
 }
@@ -590,6 +599,10 @@ func (pt *Path) colorizePath(root string, elements []string) string {
 			cycle = append(cycle[1:], cycle[0])
 		}()
 		return fmt.Sprintf("<%s>%s</>", cycle[0], element)
+	}
+
+	if len(elements) == 0 {
+		return colorizeElement(root)
 	}
 
 	colorizeSeparator := func() string {

--- a/src/segments/path_test.go
+++ b/src/segments/path_test.go
@@ -198,6 +198,14 @@ func TestAgnosterPathStyles(t *testing.T) {
 			PathSeparator:       "\\",
 			FolderSeparatorIcon: " > ",
 		},
+		{
+			Style:               Unique,
+			Expected:            "a",
+			HomePath:            homeDir,
+			Pwd:                 "/ab",
+			PathSeparator:       "/",
+			FolderSeparatorIcon: " > ",
+		},
 
 		{
 			Style:               Powerlevel,
@@ -348,6 +356,14 @@ func TestAgnosterPathStyles(t *testing.T) {
 			Pwd:                 homeDirWindows + "\\something\\man",
 			GOOS:                platform.WINDOWS,
 			PathSeparator:       "\\",
+			FolderSeparatorIcon: " > ",
+		},
+		{
+			Style:               Letter,
+			Expected:            "w",
+			HomePath:            homeDir,
+			Pwd:                 "/whatever",
+			PathSeparator:       "/",
 			FolderSeparatorIcon: " > ",
 		},
 
@@ -1091,6 +1107,13 @@ func TestAgnosterPath(t *testing.T) {
 			PathSeparator:  "/",
 			Cycle:          []string{"blue", "yellow"},
 			ColorSeparator: true,
+		},
+		{
+			Case:          "Unix one level",
+			Expected:      "mnt",
+			Home:          homeDir,
+			PWD:           "/mnt",
+			PathSeparator: "/",
 		},
 	}
 

